### PR TITLE
MM-3190 show user limit modal when cloud limit exceed

### PR DESCRIPTION
**Pull request by pablovelezvidal**
#### Summary
Shows the limit user modal to admin when they click the invite users in the welcome text channel and there are no more free seats.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-31905

#### Screenshots
upgradeModal4inviteusers

### CODE REVIEW

**Review by marianunez — CHANGES_REQUESTED** [review]
Given that we may be adding entry points to the Invite Users screens in other places in the near future, does it make sense to move the logic of checking for the limit and showing the upgrade modal within the actual invite modal? This way we avoid duplicating this logic whenever we need to open the Invite People screen

**Review by stevemudie — APPROVED** [review]
Looks good!

**Comment by nickmisasi** [comment]
> Given that we may be adding entry points to the Invite Users screens in other places in the near future, does it make sense to move the logic of checking for the limit and showing the upgrade modal within the actual invite modal? This way we avoid duplicating this logic whenever we need to open the Invite People screen

+1

**Comment by pablovelezvidal** [comment]
> > Given that we may be adding entry points to the Invite Users screens in other places in the near future, does it make sense to move the logic of checking for the limit and showing the upgrade modal within the actual invite modal? This way we avoid duplicating this logic whenever we need to open the Invite People screen
> 
> +1

yes, that was the latest update I did to the PR. @nickmisasi , @marianunez could you please take a look again?

File: components/invitation_modal/index.js [hunk 1]
```diff
@@ -50,16 +51,19 @@ export function mapStateToProps(state) {
     const isLicensed = license && license.IsLicensed === 'true';
     const isGroupConstrained = Boolean(currentTeam.group_constrained);
     const canInviteGuests = !isGroupConstrained && isLicensed && guestAccountsEnabled && haveITeamPermission(state, {team: currentTeam.id, permission: Permissions.INVITE_GUEST});
+    const isCloud = license.Cloud === 'true';
+    const freeTierWithNoFreeSeats = isCloud && subscriptionStats && subscriptionStats.is_paid_tier === 'false' && subscriptionStats.remaining_seats <= 0;
```
Comment (nickmisasi) [inline components/invitation_modal/index.js:55]: Is it possible subscriptionStats might be empty here? true && {} (if subscriptionStats is empty) would mean that the next case (subscriptionStats.is_paid_tier) access would throw an error?

File: components/invitation_modal/invitation_modal.jsx [hunk 2]
```diff
@@ -231,90 +234,104 @@ export default class InvitationModal extends React.PureComponent {
...
+                    }
+                </div>
+            </FullScreenModal>
+        </RootPortal>);
+
+        const actionsUserLimitModal = {
+            closeModal: this.props.actions.closeModal,
+            openModal: this.props.actions.openModal,
+        };
+        const userLimitModal = (<>
```
Comment (nickmisasi) [inline components/invitation_modal/invitation_modal.jsx:325]: Do you need the <> here?

File: components/invitation_modal/index.js [hunk 3]
```diff
@@ -50,16 +51,19 @@ export function mapStateToProps(state) {
...
 
     const canAddUsers = haveITeamPermission(state, {team: currentTeam.id, permission: Permissions.ADD_USER_TO_TEAM});
     return {
         invitableChannels,
         currentTeam,
         canInviteGuests,
         canAddUsers,
+        isFreeTierWithNoFreeSeats: freeTierWithNoFreeSeats,
```
Comment (nickmisasi) [inline components/invitation_modal/index.js:63]:
```suggestion
    const isFreeTierWithNoFreeSeats = isCloud && subscriptionStats && subscriptionStats.is_paid_tier === 'false' && subscriptionStats.remaining_seats <= 0;

    const canAddUsers = haveITeamPermission(state, {team: currentTeam.id, permission: Permissions.ADD_USER_TO_TEAM});
    return {
        invitableChannels,
        currentTeam,
        canInviteGuests,
        canAddUsers,
        isFreeTierWithNoFreeSeats,
```

Comment (nickmisasi) [reply components/invitation_modal/index.js:63]: nit

**Review by nickmisasi — CHANGES_REQUESTED** [review]

File: components/invitation_modal/index.js (hunk 1 above)
Comment (pablovelezvidal) [reply components/invitation_modal/index.js:55]: thanks @nickmisasi , good catch. I will validate using the isEmpty lodash function.

File: components/invitation_modal/invitation_modal.jsx (hunk 2 above)
Comment (pablovelezvidal) [reply components/invitation_modal/invitation_modal.jsx:325]: Not necessary. Removing it to get cleaner code.

**Review by marianunez — APPROVED** [review]
Great work refactoring the `UserLimitModal` directly into the `InvitationModal`! This will reduce code duplication moving forward. 

We should remove any current instances where we currently open the `UserLimitModal`. 0/5 if is in this or another PR

**Review by stevemudie — APPROVED** [review]
Tested and Passed

The link to invite users at the bottom of the Town Square channel opens the 'You've reached the user limit' modal as expected. The modal also worked and is dismissed in all fashions as you would expect.

I did notice that on first use the share links appear for a flash before going away and the modal showing up. It only happens on the first click after opening the Spinwick but after speaking with @pablovelezvidal about it he's aware and he assured me that it won't be a problem once merged so this one is good to go!

**Comment by pablovelezvidal** [comment]
> Tested and Passed
> 
> The link to invite users at the bottom of the Town Square channel opens the 'You've reached the user limit' modal as expected. The modal also worked and is dismissed in all fashions as you would expect.
> 
> I did notice that on first use the share links appear for a flash before going away and the modal showing up. It only happens on the first click after opening the Spinwick but after speaking with @pablovelezvidal about it he's aware and he assured me that it won't be a problem once merged so this one is good to go!

To clarify what @stevemudie mentions, the PR that contains the necessary code so the flash don't happen is this one https://github.com/mattermost/mattermost-webapp/pull/7468, there is the code where we load the subscriptionStats on app initial boot.

**Review by nickmisasi — APPROVED** [review]

**Pull request merged**